FreeRTOS/FreeRTOS
# Added Safe interrupt demo

Description
-----------
This PR adds a demo demonstrating best practice for safe interrupt handling by deferring work from privileged IRQ handler to unprivileged user tasks. It contains two interrupts: GPIO pin interrupt connected to a button and UART interrupt. Privileged IRQ handlers enqueue a message to a shared queue. Unprivileged user tasks receives the message from the queue and further processes the IRQ event.G

Test Steps
-----------
Built in MCUXpresso and tested on NXP LPC55S69-EVK board. Pressing the button changes LED color and UART command line interface works as expected.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
